### PR TITLE
statistics: Update stats meta count and modify count for dropped table partitions

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/48182

Problem Summary:

### What changed and how does it work?

Based on [this design doc](https://pingcap.feishu.cn/docx/OLCrdxHj2ojb9KxlUvUch60LnBe#part-PRTmdkF0doiZSxxHHmDckfeJnZf), we will only update the count and modfiy_count for it. Then it will use the count and modify_count to determine if we need to trigger an auto-analysis task.

So in this PR, I updated the count and modify count of the global table.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] [Manual test](https://pingcap.feishu.cn/docx/XPYIdC4Ihoc7jNxuPficBbOgn8g)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/statistics/handle/ddl/ddl.go [hunk 1]
```diff
@@ -116,13 +118,36 @@ func (h *ddlHandlerImpl) HandleDDLEvent(t *util.DDLEvent) error {
...
-		_, droppedPartitionInfo := t.GetDropPartitionInfo()
+		globalTableInfo, droppedPartitionInfo := t.GetDropPartitionInfo()
+
+		delta := int64(0)
+		count := int64(0)
 		for _, def := range droppedPartitionInfo.Definitions {
+			// Get the count and modify count of the partition.
+			stats := h.statsHandler.GetPartitionStats(globalTableInfo, def.ID)
+			if stats.Pseudo {
+				logutil.BgLogger().Warn(
```
Comment (hawkingrei) [inline pkg/statistics/handle/ddl/ddl.go:129]: You should use ```StatsLogger```.

**Review by qw4990 — APPROVED** [review]

File: pkg/statistics/handle/ddl/ddl.go [hunk 2]
```diff
@@ -116,13 +118,36 @@ func (h *ddlHandlerImpl) HandleDDLEvent(t *util.DDLEvent) error {
 			}
 		}
 	case model.ActionDropTablePartition:
-		// TODO: Update the modify count and count for the global table.
-		_, droppedPartitionInfo := t.GetDropPartitionInfo()
+		globalTableInfo, droppedPartitionInfo := t.GetDropPartitionInfo()
+
+		delta := int64(0)
+		count := int64(0)
```
Comment (time-and-fate) [inline pkg/statistics/handle/ddl/ddl.go:124]: Here it's always `delta = -count`, so I think we can use one variable here.

File: pkg/statistics/handle/ddl/ddl.go [hunk 3]
```diff
@@ -116,13 +118,36 @@ func (h *ddlHandlerImpl) HandleDDLEvent(t *util.DDLEvent) error {
...
 		for _, def := range droppedPartitionInfo.Definitions {
+			// Get the count and modify count of the partition.
+			stats := h.statsHandler.GetPartitionStats(globalTableInfo, def.ID)
+			if stats.Pseudo {
+				logutil.StatsLogger.Warn(
+					"drop partition with pseudo stats, "+
+						"usually it won't happen because we always load stats when initializing the handle",
+					zap.String("table", globalTableInfo.Name.O),
+					zap.String("partition", def.Name.O),
```
Comment (time-and-fate) [inline pkg/statistics/handle/ddl/ddl.go:133]: I think db name should also be printed, but looks like it would take much extra work to implement that.

**Review by time-and-fate — APPROVED** [review]
Not big problems. I'm OK to merge it now.

**Comment by time-and-fate** [comment]
Feel free to /unhold it.

File: pkg/statistics/handle/ddl/ddl.go (hunk 2 above)
Comment (hi-rustin) [reply pkg/statistics/handle/ddl/ddl.go]: Good cacth!

File: pkg/statistics/handle/ddl/ddl.go (hunk 3 above)
Comment (hi-rustin) [reply pkg/statistics/handle/ddl/ddl.go:132]: We need to pass the info schema here. And it involves a lot of test changes. I will do it in a separate PR.

Comment (hi-rustin) [reply pkg/statistics/handle/ddl/ddl.go:132]: See https://github.com/pingcap/tidb/pull/49011

**Pull request merged**